docs/versioning.md

```diff
@@ -0,0 +1,48 @@
+# Versioning and Branching in AWS Load Balancer Operator
+
+The AWS Load Balancer Operator follows the semantic versioning, for any given release `X.Y.Z`:
+* an X (major) release indicates a set of backwards-compatible changes. Changing X means there's a breaking change.
+* a Y (minor) release indicates a minimum feature set. Changing Y means the addition of a backwards-compatible feature.
+* a Z (patch) release indicates minimum set of bugfixes. Changing Z means a backwards-compatible change that doesn't add functionality.
+
+## Branches
+
+The AWS Load Balancer Operator repository contains two types of branches: the `main` branch and `release-X.Y` branches.
+
+The main branch is where development happens. All the latest code, including breaking changes, happens on `main`.
+The `release-X.Y` branches contain stable, backwards compatible code. Every minor (`X.Y`) release, a new such branch is created.
+
+## Channels
+
+The AWS Load Balancer Operator's releases get published in two types of [OLM channels](https://olm.operatorframework.io/docs/glossary/#channel): the minor `release-vX.Y` and the major `release-vX`.
+
+The minor channels contain patch releases. The major channels contain all patch releases from all minor channels.
+
+## OpenShift Version Compatibility
+
+The table below illustrates the OpenShift versions for which various AWS Load Balancer Operator releases were intended:
+
+| OCP version | AWS Load Balancer Operator branch  | AWS Load Balancer Operator OLM channel |
+| :---------: | :-------------------------------:  | :------------------------------------: |
+| 4.16        | release-1.1                        | stable-v1.1, stable-v1                 |
```

Comment:
Can you add 4.17 while we are still working on this PR?

Reply:
I will add it after the controller bump (`2.8.2`) along with a new olm channel. Like [I did in ExtDNSOperator](https://github.com/openshift/external-dns-operator/pull/233/files#diff-c08747565b40689f0a65a25b5d6e77bc49565ed7cef1a6d60e1f2c3d4ed4a8ff).